common/workunit/workunit.cpp

```diff
@@ -8427,9 +8758,81 @@ void CLocalWorkUnit::addProcess(const char *type, const char *instance, unsigned
         node = node->addPropTree(instance);
         node->setProp("@log", log);
         node->setPropInt("@pid", pid);
+        if (max > 0)
+            node->setPropInt("@max", max);
+        if (!isEmptyString(pattern))
+            node->setProp("@pattern", pattern);
     }
 }
 
+void CLocalWorkUnit::setProcessLogPattern(const char *type, const char *instance, const char *pattern)
+{
+    VStringBuffer xpath("Process/%s/%s", type, instance);
+    CriticalBlock block(crit);
+    IPropertyTree *proc = p->queryPropTree(xpath);
+    if (proc)
+        proc->setProp("@pattern", pattern);
+}
+
+StringBuffer &CLocalWorkUnit::getProcessLogPattern(const char *type, const char *instance, StringBuffer &pattern) const
+{
+    VStringBuffer xpath("Process/%s/%s", type, instance);
+    CriticalBlock block(crit);
+    IPropertyTree *proc = p->queryPropTree(xpath);
+    if (proc)
+        proc->getProp("@pattern", pattern);
+    return pattern;
+}
+
+StringBuffer &CLocalWorkUnit::getSlaveLogFileName(const char *thorProcess, int slaveNum,
+    const char *ipAddress, StringBuffer &logFileName, bool withPath) const
+{
+    CriticalBlock block(crit);
+    Owned<IPropertyTreeIterator> procs = getProcesses("Thor", thorProcess);
+    if (!procs->first())
+        return logFileName;
+
+    StringBuffer logSpec;
+    IPropertyTree &proc = procs->query();
+    proc.getProp("@log", logSpec);
+    if (logSpec.isEmpty())
+        return logFileName;
+
+    if (withPath)
+    {
+        splitFilename(logSpec, nullptr, &logFileName, nullptr, nullptr);
+        addPathSepChar(logFileName);
+    }
+
+    if (proc.hasProp("@pattern"))
+    {
+        proc.getProp("@pattern", logFileName);
+
+        StringBuffer slaveNumStr;
+        slaveNumStr.append(slaveNum);
+        logFileName.replaceString(SLAVEIDSTR, slaveNumStr);
+    }
+    else
+    { //legacy WU and not from import
+        if (slaveNum > 0)
+            logFileName.appendf("thorslave.%d", slaveNum);
+        else
+        {
+            //legacy wuid: an IP address inside the name of thor slave.
+            //ex. thorslave.10.239.219.6_20100.2012_05_23.log
+            logFileName.appendf("thorslave.%s*", ipAddress);
+        }
+
+        //Now, add the 'yyyy_mm_dd' (10 bytes) into the logFileName.
+        StringBuffer logDate;
+        getThorLogDateFromLogFileName(logSpec, "/thormaster.", logDate);
```

Comment:
related to other discussion:
feels like getSlaveLogFileName should getting a IConstWUThorLogInfo for the correct process,
then asking it for a thor slave log given a slave number.

Reply:
Will do.